src/util/dependent-resource.ts

```diff
@@ -82,16 +94,17 @@ export function buildDependentResource<T extends DependentResource>(
 			key: string,
 		): Promise<string | undefined> {
 			const id = await getResourceId(parentParam);
-			const [result] = await pine.get<DependentResource>({
+			const [result] = (await pine.get({
 				resource: resourceName,
 				options: {
 					$select: 'value',
 					$filter: {
 						[parentResourceName]: id,
 						[resourceKeyField]: key,
 					},
-				},
-			});
+				} as ODataOptionsWithoutCount<BalenaModel[T]['Read']>,
+			})) as unknown as Array<{ value: string } | undefined>;
```

Comment:
This can reduce the casts v to the very minimum:
```suggestion
			const [result] = (await pine.get({
				resource: resourceName as DependentResource,
				options: {
					$select: 'value',
					$filter: {
						[parentResourceName]: id,
						[resourceKeyField]: key,
					} as ODataOptionsWithoutCount<BalenaModel[DependentResource]['Read']>['$filter'],
				},
			}));
```
the `as DependentResource` is avoiding the `T can be any type that extends DependentResource`, but it shouldn't matter since atm DependentResource is a union of very specific strings.

Reply:
somehow it looks like this is also working with `satisfies`... but I don't trust that it's correct tbh